src/Symfony/Component/Message/README.md

```diff
@@ -0,0 +1,263 @@
+Message Component
+=================
+
+The Message component helps application to send and receive messages to/from other applications or via
+message queues.
+
+Resources
+---------
+
+  * [Contributing](https://symfony.com/doc/current/contributing/index.html)
+  * [Report issues](https://github.com/symfony/symfony/issues) and
+    [send Pull Requests](https://github.com/symfony/symfony/pulls)
+    in the [main Symfony repository](https://github.com/symfony/symfony)
+
+
+Documentation
+-------------
+
+**Note:** this documentation is to be moved to symfony.com when merging the Component.
+
+### Concepts
+
+![Component overview](Resources/doc/component-overview.png)
+
+1. **Sender**
+   Responsible for serializing and sending the message to _something_. This something can be a message broker or a 3rd 
+   party API for example.
+
+2. **Receiver**
+   Responsible for deserializing and forwarding the messages to handler(s). This can be a message queue puller or an API
+   endpoint for example.
+   
+3. **Handler**
+   Given a received message, contains the user business logic related to the message. In practice, that is just a PHP
+   callable.
+   
+
+### Bus
+
+The bus is used to dispatch messages. MessageBus' behaviour is in its ordered middleware stack. When using
+the message bus with Symfony's FrameworkBundle, the following middlewares are configured for you:
+
+1. `LoggingMiddleware` (log the processing of your messages)
+2. `SendMessageMiddleware` (enable asynchronous processing)
+3. `HandleMessageMiddleware` (call the registered handle)
+
+```php
+use App\Message\MyMessage;
+
+$result = $this->get('message_bus')->handle(new MyMessage(/* ... */));
+```
+
+### Handlers
+
+Once dispatched to the bus, messages will be handled by a "message handler". A message handler is a PHP callable 
+(i.e. a function or an instance of a class) that will do the required processing for your message. It _might_ return a
```

Comment:
I do not think this is a good idea. The two reasons that directly comes to my mind are: 

- Consider CQRS, then this is the "Command" and commands should not return anything.
- The controller/caller does not know if the message is handled async or not.

Reply:
It's not only for CQRS and that' the reason that this is named Message and not a Command.

Reply:
That also resonate with @aRn0D's question and [my answer](https://github.com/symfony/symfony/pull/24411#discussion_r143650179).

Reply:
AMQP allows messages to be acknowledged, rejected or re-queued. It is not clear how I can control it.